core/backend/src/IModelSchemaLoader.ts

```diff
@@ -145,22 +144,15 @@ class IModelSchemaLocater implements ISchemaLocaterType {
     return val.result;
   }
 
-  /** Read schema data from the iModel as JSON string
+  /** Read schema data from the iModel as JSON string asynchronously
    * @param schemaName A string with the name of the schema to load.
    * @returns A string with the JSON for the schema or `undefined` if the schema is not found.
-   * @throws [IModelError]($common) if the schema exists, but cannot be loaded.
    */
   private async getSchemaStringAsync(schemaName: string): Promise<string | undefined> {
-    const val = await new Promise<IModelJsNative.ErrorStatusOrResult<any, string>>((resolve) => {
-      this._iModel.nativeDb.getSchemaAsync(schemaName, resolve);
-    });
-
-    if (undefined !== val.error) {
-      if (IModelStatus.NotFound === val.error.status) {
-        return undefined;
-      }
-      throw new IModelError(val.error.status, `reading schema=${schemaName}`);
+    try {
```

Comment:
why not just let this throw an error so the caller has a chance of figuring out what's wrong?

Reply:
There's no point in catching an exception and then rethrowing it. This method is useless.